dsp-user/interface/src/main/scala/dsp/user/route/UserRoutes.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.user.route
+
+import zhttp.http._
+import zio._
+
+import dsp.config.AppConfig
+import dsp.errors.ValidationException
+import dsp.user.handler.UserHandler
+
+/**
+ * The UserRoutes case class which needs an instance of a userHandler
+ */
+final case class UserRoutes(userHandler: UserHandler) {
+
+  /**
+   * The user related routes which need AppConfig in the environment
+   */
+  val routes: Http[AppConfig, ValidationException, Request, Response] = Http.collectZIO[Request] {
+    // POST /admin/users
+    case req @ (Method.POST -> !! / "admin" / "users") =>
+      CreateUser
+        .route(req, userHandler)
+        .catchAll { e =>
+          ZIO.succeed(Response.text(e.getMessage).setStatus(Status.BadRequest))
```

Comment:
just wondering: in case of a failure, do we really want to return text, or rather JSON? If so, we'd probably need an object that holds a failure state, and can serialize itself to JSON

Reply:
What do you mean by "failure state"? The HTTP status code is returned, isn't that enough?

Reply:
In the long run the api should return a common type of response for all 400 errors. This is a question for designing the api. For the task at hand I think the existing solution is fine.

@BalduinLandolt I like this approach and it is not overthinking it is necessary design work we need to do for the v3 API at some point. However, exposing a trace in a response is helpful when developing but could be an unnecessary attack vector in production.

Reply:
Alright, thank you both for your inputs. I leave it for now and we can introduce such an ErrorResponse type in the future.